routing/missioncontrol.go

```diff
@@ -260,11 +261,58 @@ func (m *MissionControl) setLastPairResult(fromNode,
 		m.lastPairResult[fromNode] = nodePairs
 	}
 
-	nodePairs[toNode] = TimedPairResult{
-		Timestamp: timestamp,
-		Amt:       result.amt,
-		Success:   result.success,
+	current := nodePairs[toNode]
+
+	// Apply the new result to the existing data for this pair. If there is
+	// no existing data, apply it to the default values for TimedPairResult.
+	if result.success {
+		current.SuccessTime = timestamp
+		successAmt := result.amt
+
+		// Only update the success amount if this amount is higher. This
```

Comment:
Won't this be wrong if say you succeeded in sending 100k sats a week ago, and now you successfully send 1 sat? This will be recorded as you successfully just sent 100k sats.

Reply:
The reason to update the timestamp is mainly to get an idea of freshness for (future) prepay probe functionality. It isn't used for the probability calculation.